# `PYTHON NOTES: V25.01.08 `

## `Print() Function:`

In [3]:
print("Hello Pratik")

Hello Pratik


## `Input() Function:`

In [44]:
value = input("Enter a Value: ")
print(value)

Enter a Value:  23


23


## `Type() Function:`

In [49]:
print(type(24))
type(24)

<class 'int'>


int

## `Data Types: V25.01.08`

1. `Numeric Data Type`

In [4]:
print("Numeric Data Type")
print(type(23))
print(type(23.5))
print(type(2+2j))

Numeric Data Type
<class 'int'>
<class 'float'>
<class 'complex'>


2. `Sequential Data Type`

In [23]:
print("Sequential Data Type")
print(type([22,22.3,2+2j]))
print(type((22,22.3,2+2j)))
print(type(set("aabbccdd")))
print(type({"name":"Pratik" ,"age":22}))

Sequential Data Type
<class 'list'>
<class 'tuple'>
<class 'set'>
<class 'dict'>


3. `String Data Type`

In [6]:
print("String Data Type")
print(type("Pratik"))

<class 'str'>


4. `Boolean Data Type`

In [7]:
print("Boolean Data Type")
print(type(True))

<class 'bool'>


## Operators: V25.01.09 -  V25.01.10

1. `Assignment Arithmatic Operator`

In [2]:
print("Assignment Arithmatic Operator")
num1 = int(input("Enter 1st Number: "))
num2 = int(input("Enter 2nd Number: "))
num1+=num2
print("Addition of two Number is: ",num1)
num1-=num2
print("Subtraction of two Number is: ",num1)
num1*=num2
print("Multiplication of two Number is: ",num1)
num1/=num2
print("Division of two Number is: ",num1)
num1//=num2
print("Floor Division of two Number is: ",num1)
num1**=num2
print("Power of two Number is: ",num1)

Assignment Arithmatic Operator
Addition of two Number is:  4
Subtraction of two Number is:  2
Multiplication of two Number is:  4
Division of two Number is:  2.0
Floor Division of two Number is:  1.0
Power of two Number is:  1.0


2. `Relational Operator`

In [28]:
print("Relational Operator")
print(num1>=num2)
print(num1<=num2)
print(num1==num2)
print(num1!=num2)

Enter 1st Number:  7
Enter 2nd Number:  5


True
False
True
False
False
True


3. `Logical Operator V25.01.10`

In [1]:
print("Logical Operator")
print(44>=33 and 44>=22)
print(22<=22 or 33<=33)
print(not True)

Logical Operator
True
True
False


4. `Bitwise Operator`

In [46]:
print("Bitwise Operator")
print(bin(15))
print(1 & 1)
print(15 | 11)
print(15 ^ 11)
print(~11)
print(bin(15>>2))
print(bin(15<<2))
print(15>>2)
print(15<<2)

0b1111
11
15
4
-12
0b11
0b111100
3
60


5. `Membership Operator`

In [2]:
print("Membership Operator:")
a = "ABCD"
print("A" in a)

Membership Operator:
True


6. `Identity Operator`

In [4]:
print("Identity Operator:")
a = "ABCD"
b = "ABCD"
print(a is b)

Identity Operator:
True


## `Jupyter Installation: V25.01.13`

## `Control Statement: V25.01.14`

1. `IF Statement`

In [4]:
if True:
    print("Welcome Guys!!")

Welcome Guys!!


In [9]:
if False:
    print("Welcome Guys!!")
    # Blanck!!  Print Nothing.

In [10]:
if 1:
    print("Welcome Guys!!")
    # Truth Value

Welcome Guys!!


In [14]:
if 0:
    print("Welcome Guys!!")
    # Blanck!!  Print Nothing. Falsy Value

In [16]:
if -0:
    print("Welcome Guys!!")
    # Blanck!!  
    # Print Nothing. 
    # Falsy Value
    # Entity : Something that exist.
    # -0 : Technicaly Wrong. 

2. `If-Else Statement`

In [26]:
if True:
    print("Welcome Guys!!")
else:
    print("Bhaag...")

Welcome Guys!!


- `Question:` Accept a no from user and check if it is Even or Odd.

In [22]:
print("Print Even or Odd Numbers.")
num = int(input("Enter a Number: "))
if num%2 == 0 :
    print(num," is a Even Number.")
else:
    print(num," is a Odd Number.")

Print Even or Odd Numbers.


Enter a Number:  5


5  is a Odd Number.


- `Question:` Accept a no from user and check if it is divisible by 7 or 13.

In [25]:
print("Check a number divisible by 7 or 13.")
num = int(input("Enter a Number: "))
if num%7 == 0 or num%13 ==0 :
    print(num," is divisible.")
else:
    print(num," is not divisible.")

Check a number divisible by 7 or 13.


Enter a Number:  14


14  is divisible.


3. If-Elif Statement | If-Elif Ladder

- `Question:` Accept a no from user and check if it is Greater, Smaller or Equal to 23. 

In [30]:
print("Check a Number is Greater, Smaller or Equal.")
num = int(input("Enter a Number: "))
if num>23:
    print(num," is a Greater than 23.")
elif num==23:
    print(num," is a Equal to 23.")
else:
    print(num," is a Smaller than 23.")

Check a Number is Greater, Smaller or Equal.


Enter a Number:  23


23  is a Equal to 23.


- `Question:` Accept a Budget from User. 

In [35]:
budget = int(input("Enter your Budget: "))
if budget>=300 and budget<=500:
    print("International Trip.")
elif budget>=200 and budget<=299:
    print("National Trip.")
elif budget>=100 and budget<=199:
    print("City Trip.")
elif budget>=50 and budget<=99:
    print("Local Trip.")
elif budget>=1 and budget<=49:
    print("Gully Trip.")
else:
    print("Undefine Budget.")

Enter your Budget:  1


Gully Trip.


- `Question:` Accept a Designation from User and print is they apply for senior Position. 

In [2]:
designation = input("Enter what Designationyou are Applying? Manager or Executive")
if "Manager" == designation:
    senior = input("Are you  applying for Senior Position? Yes or No:")
    if "yes" == senior:
        print("2Lakh Sallary.")
    else:
        print("1Lakh Sallary.")
elif "Executive" == designation:
    senior = input("Are you applying for Senior Position? Yes or No:")
    if "yes" == senior:
        print("70K Sallary.")
    else:
        print("25K Sallary.")
else:
    print("Undefined Designation.")

Enter what Designationyou are Applying? Manager or Executive Executive
Are you applying for Senior Position? Yes or No: No


25K Sallary.


4. `Match Case`

In [15]:
value = 3
match value:
    case 1:
        print("Label 1")
    case 2:
        print("Label 2")
    case 3:
        print("Label 3")
    case _:
        print("Default Case ")

Label 3


## `Loops: V25.01.15`

### `For Loop - Repetition`

In [7]:
for i in range(1,10):
    print(i)

1
2
3
4
5
6
7
8
9


In [5]:
for i in range(1,11):
    print(i)

1
2
3
4
5
6
7
8
9
10


In [14]:
for i in range(1,11):
    print(i, end=" ")

1 2 3 4 5 6 7 8 9 10 

In [13]:
for i in range(20,41):
    print(i, end=" ")

20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 

- `Question: Print Even Numbers 1 to 100`

In [12]:
for i in range(1,101):
    if i%2==0:
        print(i, end=" ")

2 4 6 8 10 12 14 16 18 20 22 24 26 28 30 32 34 36 38 40 42 44 46 48 50 52 54 56 58 60 62 64 66 68 70 72 74 76 78 80 82 84 86 88 90 92 94 96 98 100 

- `Question: Display Number Divisible by 2 & 5 from 1 to 100.`

In [19]:
for i in range(1, 101):
    if i%2==0 and i%5==0:
        print(i, end=" ")c

10 20 30 40 50 60 70 80 90 100 

- `Question: Display Number Divisible by 13,17 from 1 to 100.`

In [19]:
for i in range(1, 101):
    if i%13==0 and i%17==0:
        print(i, end=" ")c

10 20 30 40 50 60 70 80 90 100 

- `Patterns:`

- `Square Row Number Patterns:`
- `11111`
- `22222`
- `33333`
- `44444`
- `55555`

In [20]:
for i in range(1,6):
    for j in range(1,6):
        print(i, end=" ")
    print("")

1 1 1 1 1 
2 2 2 2 2 
3 3 3 3 3 
4 4 4 4 4 
5 5 5 5 5 


- `Square Column Number Patterns:`
- `12345`
- `12345`
- `12345`
- `12345`
- `12345`

In [21]:
for i in range(1,6):
    for j in range(1,6):
        print(j, end=" ")
    print("")

1 2 3 4 5 
1 2 3 4 5 
1 2 3 4 5 
1 2 3 4 5 
1 2 3 4 5 


- `Right Half Increasing Pyramid Column Number Patterns:`
- `1`
- `12`
- `123`
- `1234`
- `12345`

In [24]:
for i in range(1,6):
    for j in range(1,6):
        if i>=j:
            print(j, end=" ")
    print("")

1 
1 2 
1 2 3 
1 2 3 4 
1 2 3 4 5 


- `Right Half Decreasing Pyramid Column Number Patterns:`
- `12345`
- `1234`
- `123`
- `12`
- `1`

In [62]:
for i in range(1, 6):
    for j in range(1, 6):
        print(j, end=" ")
    print("")

1 2 3 4 5 
1 2 3 4 5 
1 2 3 4 5 
1 2 3 4 5 
1 2 3 4 5 


- `Decreasing Pyramid Patterns:`
- `12345`
- `2345`
- `345`
- `45`
- `5`

In [25]:
for i in range(1,6):
    for j in range(1,6):
        if i<=j:
            print(j, end=" ")
    print("")

1 2 3 4 5 
2 3 4 5 
3 4 5 
4 5 
5 


- `Star Square Patterns:`
- `* * * * *`
- `* * * * *`
- `* * * * *`
- `* * * * *`
- `* * * * *`

In [26]:
for i in range(1,6):
    for j in range(1,6):
        print("*", end=" ")
    print("")

* * * * * 
* * * * * 
* * * * * 
* * * * * 
* * * * * 


- `Right Half Pyramid Patterns:`
- `*`
- `* *`
- `* * *`
- `* * * *`
- `* * * * *`

In [27]:
for i in range(1,6):
    for j in range(1,6):
        if i>=j:
            print("*", end=" ")
    print("")

* 
* * 
* * * 
* * * * 
* * * * * 


- `Left Half Pyramid Patterns:`
- `* * * * *`
- `* * * *`
- `* * *`
- `* *`
- `*`

In [28]:
for i in range(1,6):
    for j in range(1,6):
        if i<=j:
            print("*", end=" ")
    print("")

* * * * * 
* * * * 
* * * 
* * 
* 


- `Space & Half Pyrmid Pattern` `V25.01.26`
- `    *`
- `   **`
- `  ***`
- ` ****`
- `*****`

In [6]:
for i in range(1,6):
    for j in range(5,0,-1):
        if j>=i:
            print(" ",end="")
    for k in range(1,6):
        if k<=i:
            print("*",end="")  
    print("")

     *
    **
   ***
  ****
 *****


- `Space & Half Pyrmid Pattern`
- `*****`
- ` ****`
- `  ***`
- `   **`
- `    *`

In [7]:
for i in range(1,6):
    for j in range(1,6):
        if j<=i:
            print(" ",end="")
    for k in range(5,0,-1):
        if k>=i:
            print("*",end="")  
    print("")

 *****
  ****
   ***
    **
     *


- `Space & Pyramid Pattern`
- `    **`
- `   ****`
- `  ******`
- ` ********`
- `**********`

In [10]:
for i in range(1,6):
    for j in range(5,0,-1):
        if j>=i:
            print(" ",end="")
    for k in range(1,6):
        if k<=i:
            print("*",end="")
    for k in range(1,6):
        if k<=i:
            print("*",end="")  
    print("")

     **
    ****
   ******
  ********
 **********


- `Space & Pyramid Pattern`
- `    *`
- `   ***`
- `  *****`
- ` *******`
- `*********`

In [11]:
for i in range(1,6):
    for j in range(5,0,-1):
        if j>=i:
            print(" ",end="")
    for k in range(1,6):
        if k<=i:
            print("*",end="")
    for k in range(2,6):
        if k<=i:
            print("*",end="")  
    print("")

     *
    ***
   *****
  *******
 *********


- `Space & Pyramid Pattern`
- `**********`
- ` ********`
- `  ******`
- `   ****`
- `    **`

In [12]:
for i in range(1,6):
    for j in range(1,6):
        if j<=i:
            print(" ",end="")
    for k in range(5,0,-1):
        if k>=i:
            print("*",end="")
    for k in range(5,0,-1):
        if k>=i:
            print("*",end="")  
    print("")

 **********
  ********
   ******
    ****
     **


- `Space & Pyramid Pattern`
- `*********`
- ` *******`
- `  *****`
- `   ***`
- `    *`

In [13]:
for i in range(1,6):
    for j in range(1,6):
        if j<=i:
            print(" ",end="")
    for k in range(5,0,-1):
        if k>=i:
            print("*",end="")
    for k in range(4,0,-1):
        if k>=i:
            print("*",end="")  
    print("")

 *********
  *******
   *****
    ***
     *


- `Diamond Pattern`
- `    *`
- `   ***`
- `  *****`
- ` *******`
- `*********`
- `*********`
- ` *******`
- `  *****`
- `   ***`
- `    *`

In [36]:
for i in range(1,6):
    for j in range(5,0,-1):
        if j>=i:
            print(" ",end="")
    for k in range(1,6):
        if k<=i:
            print("*",end="")
    for k in range(2,6):
        if k<=i:
            print("*",end="")  
    print("")
for i in range(1,5):
    for j in range(0,5):
        if j<=i:
            print(" ",end="")
    for k in range(4,0,-1):
        if k>=i:
            print("*",end="")
    for k in range(3,0,-1):
        if k>=i:
            print("*",end="")  
    print("")

     *
    ***
   *****
  *******
 *********
  *******
   *****
    ***
     *


## `Function: V25.01.20`

- Block of code which executes targeted Logic.

## `Non-paametrise Function`

In [7]:
def Info():
    print("Hello I.T.")
Info()

Hello I.T.


In [8]:
def Name():
    print("Hello Pratik..!")
Info()

Hello I.T.


- `return keyword` : `Returns to Function Call Value.`

In [11]:
def Data():
    a = int(input("Enter a Number: "))
    return a;
Num = Data()
print("Enter Number is: ",Num)

Enter a Number:  55


Enter Number is:  55


- `Question: Accept two values from User and return there Addition, Subtraction, Multiplication, Division.`

In [13]:
def add():
    a = int(input("Enter 1'st Number: "))
    b = int(input("Enter 2'nd Number: "))
    return a+b;
result = add()
print("Addition of two Numbers is: ",result)

Enter 1'st Number:  5
Enter 2'nd Number:  5


Addition of two Numbers is:  10


In [16]:
def sub():
    a = int(input("Enter 1'st Number: "))
    b = int(input("Enter 2'nd Number: "))
    return a-b;
result = sub()
print("Subtraction of two Numbers is: ",result)

Enter 1'st Number:  6
Enter 2'nd Number:  5


Subtraction of two Numbers is:  1


In [17]:
def mul():
    a = int(input("Enter 1'st Number: "))
    b = int(input("Enter 2'nd Number: "))
    return a*b;
result = mul()
print("Multiplication of two Numbers is: ",result)

Enter 1'st Number:  5
Enter 2'nd Number:  5


Multiplication of two Numbers is:  25


In [18]:
def div():
    a = int(input("Enter 1'st Number: "))
    b = int(input("Enter 2'nd Number: "))
    return a/b;
result = div()
print("Division of two Numbers is: ",result)

Enter 1'st Number:  5
Enter 2'nd Number:  5


Division of two Numbers is:  1.0


In [19]:
def mod():
    a = int(input("Enter 1'st Number: "))
    b = int(input("Enter 2'nd Number: "))
    return a%b;
result = mod()
print("Modulos of two Numbers is: ",result)

Enter 1'st Number:  5
Enter 2'nd Number:  5


Modulos of two Numbers is:  0


In [20]:
def expo():
    a = int(input("Enter 1'st Number: "))
    b = int(input("Enter 2'nd Number: "))
    return a**b;
result = expo()
print("Power of two Numbers is: ",result)

Enter 1'st Number:  5
Enter 2'nd Number:  2


Power of two Numbers is:  25


- `break keyword is part of loop`  -> `Terminates the Execution of Code.`

In [28]:
def loop1():
    for i in range(1,11):
        if i == 5:
            break
        else:
            print(i,end=" ")
loop1()

1 2 3 4 

- `continue keyword is part of loop` -> `Omits Perticular Situation.`

In [23]:
def loop1():
    for i in range(1,11):
        if i == 5:
            continue
        else:
            print(i,end=" ")
loop1()

1 2 3 4 6 7 8 9 10 

In [25]:
def loop1():
    for i in range(1,11):
        if i == 7:
            continue
        else:
            print(i,end=" ")
loop1()

1 2 3 4 5 6 8 9 10 

In [30]:
def loop1():
    for i in range(1,11):
        if i == 7:
            print(i,end=" ")
            continue
        else:
            print(i,end=" ")
loop1()

1 2 3 4 5 6 7 8 9 10 

In [36]:
def data():
    a = 5
    return a
print(data())

5


In [35]:
def data():
    a = 5
    return a * 5
print(data())

25


In [39]:
def data1():
    a = 5
    return a * 5
def data2():
    a = "Pratik"
    return a
print("Value of Int is:",data1(),"Value of String is:",data2())
# Single Python Code has Mutiple Funtions.
# function name should be Unique.
# Multiple Python Function can be call in single print function.

Value of Int is: 25 Value of String is: Pratik


### `Parametrise Function` 
- `Compatible - Same Data Type`
- `No. of Argument should Match with No. of Parameters. `

In [43]:
def add(a,b,c):
    return a+b+c
print("Addition of two Number is",add(12,13,14))

Addition of two Number is 39


In [44]:
def add(a,b,c):
    return a+b+c
# print("Addition of two Number is",add(12,13,14))
def add(a,b,c):
    return a+b+c
print("Addition of two Number is",add(12,13,14))

Addition of two Number is 39


In [48]:
def add(a,b):
    return a+b+c
# print("Addition of two Number is",add(12,13,14))
def add(a,b,c):
    return a+b+c
print("Addition of two Number is",add(12,13,14))
# Processor Executes Closer Function. jo sabse close hoga use execute karega. means here is : def add(a,b,c):

Addition of two Number is 39


In [49]:
def add(a,b):
    print("Block 1")
    return a+b+c
# print("Addition of two Number is",add(12,13,14))
def add(a,b,c):
    print("Block 2")
    return a+b+c
print("Addition of two Number is",add(12,13,14))
# Processor Executes Closer Function. jo sabse close hoga use execute karega. means here is : def add(a,b,c):

Block 2
Addition of two Number is 39


In [47]:
def add(a,b):
    print("Block 1")
    return a+b+c
# print("Addition of two Number is",add(12,13,14))
def add(a,b,c):
    print("Block 2")
    return a+b+c
print("Addition of two Number is",add(12,13))
# Creates Ambiguity : Multiple Functions has same name. Mis-match Function. priority has closest one Function.  
# Processor Executes Closer Function. jo sabse close hoga use execute karega. means here is : def add(a,b):

TypeError: add() missing 1 required positional argument: 'c'

- `Arithmatic Operatios`

In [52]:
def add(num1,num2):
    return num1+num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
print("Addition of two Number is ",add(num1,num2))

Enter 1'st Number:  5
Enter 2nd Number:  5


Addition of two Number is  25


In [53]:
def sub(num1,num2):
    return num1-num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
print("Subtraction of two Number is ",sub(num1,num2))

Enter 1'st Number:  6
Enter 2nd Number:  5


Subtraction of two Number is  -1


In [56]:
def mul(num1,num2):
    return num1*num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
print("Product of two Number is ",sub(num1,num2))

Enter 1'st Number:  5
Enter 2nd Number:  5


Multiplication of two Number is  25


In [57]:
def div(num1,num2):
    return num1/num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
print("Questiont of two Number is ",div(num1,num2))

Enter 1'st Number:  5
Enter 2nd Number:  5


Divesion of two Number is  1.0


In [58]:
def floor(num1,num2):
    return num1//num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
print("Base Integer value of two Number is ",floor(num1,num2))

Enter 1'st Number:  5
Enter 2nd Number:  5


Floor of two Number is  1


In [60]:
def mod(num1,num2):
    return num1%num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
print("Remainder of two Number is ",mod(num1,num2))

Enter 1'st Number:  6
Enter 2nd Number:  6


Remainder of two Number is  0


In [62]:
def expo(num1,num2):
    return num1**num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
print("Power of two Number is ",expo(num1,num2))

Enter 1'st Number:  5
Enter 2nd Number:  3


Power of two Number is  125


- `Bitwise Operatios`

In [67]:
print("Bitwise Operator")
print(bin(15))
print(1 & 1)
print(15 | 11)
print(15 ^ 11)
print(~11)
print(bin(15>>2))
print(bin(15<<2))
print(15>>2)
print(15<<2)

Bitwise Operator
0b1111
1
15
4
-12
0b11
0b111100
3
60


In [70]:
def aand(num1,num2):
    return num1 & num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
result = aand(num1,num2)
print("And of two Number is ",result)

Enter 1'st Number:  1
Enter 2nd Number:  0


And of two Number is  0


In [71]:
def oor(num1,num2):
    return num1 | num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
result = oor(num1,num2)
print("Or of two Number is ",result)

Enter 1'st Number:  1
Enter 2nd Number:  0


Or of two Number is  1


In [72]:
def exor(num1,num2):
    return num1 ^ num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
result = exor(num1,num2)
print("Ex-Or of two Number is ",result)

Enter 1'st Number:  15
Enter 2nd Number:  12


Ex-Or of two Number is  3


In [73]:
def exor(num1):
    return ~num1;
num1 = int(input("Enter 1'st Number: "))
result = exor(num1)
print("Ex-Or of two Number is ",result)

Enter 1'st Number:  1


Ex-Or of two Number is  -2


In [74]:
def lso(num1,num2):
    return num1 << num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
result = lso(num1,num2)
print("Ex-Or of two Number is ",result)

Enter 1'st Number:  15
Enter 2nd Number:  2


Ex-Or of two Number is  60


In [75]:
def rso(num1,num2):
    return num1 >> num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
result = rso(num1,num2)
print("Ex-Or of two Number is ",result)

Enter 1'st Number:  15
Enter 2nd Number:  2


Ex-Or of two Number is  3


In [79]:
def rso(num1,num2):
    return num1 >> num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
result = rso(num1,num2)
print(bin(num1))
print("Ex-Or of two Number is ",bin(result))
print("Ex-Or of two Number is ",result)

Enter 1'st Number:  15
Enter 2nd Number:  2


0b1111
Ex-Or of two Number is  0b11
Ex-Or of two Number is  3


In [80]:
def lso(num1,num2):
    return num1 << num2;
num1 = int(input("Enter 1'st Number: "))
num2 = int(input("Enter 2nd Number: "))
result = lso(num1,num2)
print(bin(num1))
print("Ex-Or of two Number is ",bin(result))
print("Ex-Or of two Number is ",result)

Enter 1'st Number:  15
Enter 2nd Number:  2


0b1111
Ex-Or of two Number is  0b111100
Ex-Or of two Number is  60


### `OOP's: Object Oriented Programming V25.01.21`
- `Identation`
- `Class`
- `Constructor`
- `Class Constructor`
- `Method`
- `Object`
- `Refrence Variable`
- `Instance Variable`
- `Local Variable`
- `Self keyword`
- `Self parameter` 

In [2]:
class DataInformation:
    def Data(self):
        print("Python Programming!!")
obj = DataInformation()
obj.Data()        

Python Programming!!


In [4]:
class Data:
    def Method1(self):
        print("Method1")
    def Method2(self,num1):
        print("Method2")
        print("num1 is ",num1)
    def Method3(self,num1,num2):
        print("Method3")
        print("num1 is ",num1)
        print("num2 is ",num2)
obj = Data()
obj.Method1()
obj.Method2(21)
obj.Method3(31,41)

Method1
Method2
num1 is  21
Method3
num1 is  31
num2 is  41


In [8]:
a = 1
while(a<11):
    print(a,end=" ")
    a=a+1

1 2 3 4 5 6 7 8 9 10 

In [10]:
a = 10
while(a>0):
    print(a,end=" ")
    a=a-1

10 9 8 7 6 5 4 3 2 1 

In [20]:
class Iteration:
    def Loop1(self):
        a = 1
        while(a<11):
            print(a,end=" ")
            a = a+1
    print()
    def Loop2(self):
        a = 10
        while(a>0):
            print(a,end=" ")
            a = a-1
obj = Iteration()
obj.Loop1()
print()
obj.Loop2()


1 2 3 4 5 6 7 8 9 10 
10 9 8 7 6 5 4 3 2 1 

- `Armstrong Number: 153` `A Number whose Summation of Cube of Each Digit is the same Number.`

In [29]:
class Armstrong:
    def Evaluate(self):
        num = int(input("Enter a Number"))
        temp = num
        sum = 0
        while(num>0):
            rem = num % 10
            sum = sum + (rem*rem*rem)
            num = num // 10
        if sum == temp:
            print(sum,"is an Armstrong Number.")
        else:
            print(temp,"is not an Armstrong Number.")
obj = Armstrong()
obj.Evaluate()

Enter a Number 121


121 is not an Armstrong Number.


- `Palindrome Number: 121` `A Number Whose Reverce is the same Number.`

In [33]:
class Palindrome:
    def Evaluate(self):
        num = int(input("Enter a Number: "))
        temp = num
        sum = 0
        while(num>0):
            rem = num % 10
            sum = (sum *10) + rem
            num = num // 10
        if sum == temp:
            print(sum, "is a Palindrom Number.")
        else:
            print(temp, "is not a Palindrome Number.")
obj = Palindrome()
obj.Evaluate()

Enter a Number:  232


232 is a Palindrom Number.


- `Factorial Number: 5!` `Consider fact = 1 and multiply with ( i ) till the range is satisfy`

In [41]:
class Factorial:
    def Evaluate(self):
        fact = 1
        num = int(input("Enter a Number: "))
        for i in range(1,num+1):
            fact = fact*i
        print("Factorial of ",num,"is ",fact)
obj = Factorial()
obj.Evaluate()

Enter a Number:  5


Factorial of  5 is  120


- `Fibonacci Series: 0 1 1 2 3 5 8` `Addition of Privious two Numbers.` 

In [46]:
class Fibonacci:
    def Evaluate(self):
        p1 = 0
        p2 = 1
        num = int(input("Enter the Range: "))
        print(p1,"",p2," ",end="")
        for i in range(2,num):
            p3 = p1 + p2
            print(p3," ",end="")
            p1 = p2
            p2 = p3
obj = Fibonacci()
obj.Evaluate()

Enter the Range:  10


0  1  1  2  3  5  8  13  21  34  

- `Constructor`

In [47]:
class Data:
    a = 10
    b = 20
    def __init__(self):
        print("Constructor..")
Data()

Constructor..


In [70]:
class Data:
    num1 = 10
    num2 = 20
    def __init__(self):
        print("Constructor..")
    def Method1(self):
        print("Method1")
    def Method2(self,num1):
        print("Method2")
        print("num1 is: ",num1)
    def Method3(self,num1,num2):
        print("Method3")
        print("num1 is: ",num1)
        print("num2 is: ",num2)
obj = Data()
obj.Method1()
obj.Method2(30)
obj.Method3(40,50)

Constructor..
Method1
Method2
num1 is:  30
Method3
num1 is:  40
num2 is:  50


In [71]:
class Data:
    num1 = 10
    num2 = 20
    def __init__(self):
        print("Constructor..")
    def Method1(self):
        print("Method1")
    def Method2(self,num1):
        print("Method2")
        print("num1 is: ",num1)
    def Method3(self,num1,num2):
        print("Method3")
        print("num1 is: ",self.num1)
        print("num2 is: ",self.num2)
obj = Data()
obj.Method1()
obj.Method2(30)
obj.Method3(40,50)

Constructor..
Method1
Method2
num1 is:  30
Method3
num1 is:  10
num2 is:  20


In [72]:
class Data:
    num1 = 10
    num2 = 20
    def __init__(self):
        print("Constructor..")
    def Method1(self):
        self.Method2(30)
        print("Method_1")
    def Method2(self,num1):
        self.Method3(40,50)
        print("Method_2")
        print("num1 is: ",self.num1)
    def Method3(self,num1,num2):
        print("Method_3")
        print("num1 is: ",self.num1)
        print("num2 is: ",self.num2)
obj = Data()
obj.Method1()


Constructor..
Method_3
num1 is:  10
num2 is:  20
Method_2
num1 is:  10
Method_1


In [73]:
class Data:
    num1 = 10
    num2 = 20
    def __init__(self,num1,num2):
        print("Constructor..")
        print("num1 is: ",num1)
        print("num2 is: ",num2)
    def Method1(self):
        print("Method1")
    def Method2(self,num1):
        print("Method2")
        print("num1 is: ",num1)
    def Method3(self,num1,num2):
        print("Method3")
        print("num1 is: ",self.num1)
        print("num2 is: ",self.num2)
obj = Data(100,200)
obj.Method1()
obj.Method2(30)
obj.Method3(40,50)

Constructor..
num1 is:  100
num2 is:  200
Method1
Method2
num1 is:  30
Method3
num1 is:  10
num2 is:  20


In [74]:
class Data:
    num1 = 10
    num2 = 20
    def __init__(self,num1,num2):
        print("Constructor..")
        print("num1 is: ",self.num1)
        print("num2 is: ",self.num2)
    def Method1(self):
        print("Method1")
    def Method2(self,num1):
        print("Method2")
        print("num1 is: ",num1)
    def Method3(self,num1,num2):
        print("Method3")
        print("num1 is: ",self.num1)
        print("num2 is: ",self.num2)
obj = Data(100,200)
obj.Method1()
obj.Method2(30)
obj.Method3(40,50)

Constructor..
num1 is:  10
num2 is:  20
Method1
Method2
num1 is:  30
Method3
num1 is:  10
num2 is:  20


- `Lambda Function: V25.01.23` 

In [1]:
value = lambda a: a+10
print(value(25))

35


In [4]:
a = lambda c,d,e: c*d*e
print(a(5,7,9))

315


- `Nested function (Inner Function):`

In [10]:
def Add(a,b):
    print("Addition is ",(a+b))
    def Sub(a,b):
        print("Subtraction is ",(a-b))
        def Mul(a,b,c):
            print("Multiplication is ",(a*b*c))
            def Div(a,b):
                print("Division is ",(a/b))
            Div(10,5)
        Mul(10,20,30)
    Sub(10,11)
Add(5,7)

Addition is  12
Subtraction is  -1
Multiplication is  6000
Division is  2.0


### `String & It's functions:`

- `isalnum:`

In [14]:
data = "Python12345"
print(data.isalnum())

True


In [15]:
data = "Python 12345"
print(data.isalnum())

False


- `isalpha:`

In [16]:
data = "ABCD"
print(data.isalpha())

True


In [17]:
data = "1ABCD"
print(data.isalpha())

False


- `isupper:`

In [21]:
data = "ABCD"
print(data.isupper())

True


- `islower:`

In [23]:
data = "ABCD"
print(data.islower())

False


In [12]:
data = "pqrs"
print(data.islower())

True


In [11]:
data = "pqrs"
print(data.isupper())

False


- `isdigit:`

In [25]:
data = "1234"
print(data.isdigit())

True


In [13]:
data = "1234 "
print(data.isdigit())

False


In [27]:
data = "1234A"
print(data.isdigit())

False


- `Concatination:`

In [31]:
a = "hi,"
b = " How are you?"
result = a+b
print(result)

hi, How are you?


In [32]:
a = "hi,"
b = "How are you?"
result = a+" "+b
print(result)

hi, How are you?


- `join:`

In [35]:
s1 = " sad "
s2 = "Happy"
result = s1.join(s2)
print(result)

H sad a sad p sad p sad y


In [37]:
s1 = "sad"
s2 = " Happy "
result = s2.join(s1)
print(result)

s Happy a Happy d


- `lower:`

In [38]:
s3 = "HAPPY"
print(s3.lower())

happy


- `upper:`

In [39]:
s3 = "happy"
print(s3.upper())

HAPPY


In [40]:
s3 = "AbCd"
print(s3.upper())

ABCD


In [41]:
s3 = "AbCd"
print(s3.lower())

abcd


- `swapcase:`

In [42]:
s3 = "AbCd"
print(s3.swapcase())

aBcD


- `title:`

In [43]:
s3 = "coading is cool when you understand it."
print(s3.title())

Coading Is Cool When You Understand It.


- `Format Specifier:`

In [44]:
id = 731
name = "Pandu"
percentage = 73.9
print("Id is %d, name is %s, Percentage are %f"%(id,name,percentage))

Id is 731, name is Pandu, Percentage are 73.900000


In [45]:
id = 731
name = "Pandu"
percentage = 73.9
print("Id is %d, name is %s, Percentage are %f"%(name,id,percentage))

TypeError: %d format: a real number is required, not str

In [47]:
rollno = 731
name = "Pratik"
percentage = 80.0
print("Roll No is %d, Name is %s, Percentage are %f"%(rollno,name,percentage))

Roll No is 731, Name is Pratik, Percentage are 80.000000


- `replace:`

In [48]:
s3 = "Pandu is a bad boy."
print(s3.replace("bad","good"))

Pandu is a good boy.


- `find:`

In [53]:
s3 = "abcdefgh"
print(s3.find("b"))

1


In [55]:
s3 = "abcdefgh"
print(s3.find("abc"))

0


In [56]:
s3 = "abcdefgh"
print(s3.find("z"))

-1


- `index:`

In [57]:
s3 = "abcdefgh"
print(s3.index("c"))

2


In [58]:
s3 = "abcdefgh"
print(s3.index("z"))

ValueError: substring not found

## `List: V25.01.24`

- `append: Add the element at the end of the list.`

In [4]:
# append element: full element add
a = [22,23,"Python",4+9j]
a.append("hello")
print(a)

[22, 23, 'Python', (4+9j), 'hello']
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)]]


In [2]:
# append list into the another list: 
a = [22,23,"Python",4+9j]
a.append("hello")
print(a)
b = [22,23,"Python",4+9j]
a.append(b)
print(a)

[22, 23, 'Python', (4+9j), 'hello']
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)]]


In [6]:
# append tuple into the list:
a = [22,23,"Python",4+9j]
a.append("hello")
print(a)
b = [22,23,"Python",4+9j]
a.append(b)
print(a)
c = (11,22,33,44)
a.append(c)
print(a)

[22, 23, 'Python', (4+9j), 'hello']
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)]]
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)], (11, 22, 33, 44)]


- `extend: merge new iteration into the current list.`

In [8]:
# extend list: means mearge with exissting list:
a = [22,23,"Python",4+9j]
a.append("hello")
print(a)
b = [22,23,"Python",4+9j]
a.append(b)
print(a)
c = (11,22,33,44)
a.append(c)
print(a)
d2 = ["happy", "sad","nuteral"]
a.extend(d2)
print(a)

[22, 23, 'Python', (4+9j), 'hello']
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)]]
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)], (11, 22, 33, 44)]
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)], (11, 22, 33, 44), 'happy', 'sad', 'nuteral']


In [5]:
# extend with tuple: 
a = [22,23,"Python",4+9j]
a.append("hello")
print(a)
b = [22,23,"Python",4+9j]
a.append(b)
print(a)
c = (11,22,33,44)
a.append(c)
print(a)
d2 = ["happy", "sad","nuteral"]
a.extend(d2)
print(a)
d3 = (55,66,77,88)
a.extend(d3)
print(a)

[22, 23, 'Python', (4+9j), 'hello']
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)]]
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)], (11, 22, 33, 44)]
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)], (11, 22, 33, 44), 'happy', 'sad', 'nuteral']
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)], (11, 22, 33, 44), 'happy', 'sad', 'nuteral', 55, 66, 77, 88]


In [14]:
# extend element: one by one single charecter in the element added in list
a = [22,23,"Python",4+9j]
a.append("hello")
print(a)
b = [22,23,"Python",4+9j]
a.append(b)
print(a)
c = (11,22,33,44)
a.append(c)
print(a)
d2 = ["happy", "sad","nuteral"]
a.extend(d2)
print(a)
a.extend("Bye")
print(a)

[22, 23, 'Python', (4+9j), 'hello']
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)]]
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)], (11, 22, 33, 44)]
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)], (11, 22, 33, 44), 'happy', 'sad', 'nuteral']
[22, 23, 'Python', (4+9j), 'hello', [22, 23, 'Python', (4+9j)], (11, 22, 33, 44), 'happy', 'sad', 'nuteral', 'B', 'y', 'e']


- `insert: It is use to add element at specific index position inside a list.`
- `remove: It will remove 1'st occurance of the specific element.`

In [32]:
# insert: insert at specific index position
b2 = [9+8j,"Guido", "Coding", 4.7]
b2.insert(0,"Hello")
print(b2)
b2 = [9+8j,"Guido", "Coding", 4.7]
b2.insert(2,"Logic")
print(b2)
# remove: remove 1st occurance of the element
b2.remove("Guido")
print(b2)
b2.insert(3,"4.7")
print(b2)
b2.remove(4.7)
print(b2)
# pop: specific index position element will be poped out
print(b2.pop())
print(b2)
print(b2.pop(1))
print(b2)
# index: It will return the index value of the specific element
print(b2.index(9+8J))
print(b2.index("Coding"))
# count: it will return the no of occurance of the specific element
print(b2.count("Coding"))
b2.insert(1, "Coding")
b2.insert(4, "Coding")
print(b2.count("Coding"))
print(b2)
# reverse: reverse the order of your list
b2.reverse()
print(b2)

['Hello', (9+8j), 'Guido', 'Coding', 4.7]
[(9+8j), 'Guido', 'Logic', 'Coding', 4.7]
[(9+8j), 'Logic', 'Coding', 4.7]
[(9+8j), 'Logic', 'Coding', '4.7', 4.7]
[(9+8j), 'Logic', 'Coding', '4.7']
4.7
[(9+8j), 'Logic', 'Coding']
Logic
[(9+8j), 'Coding']
0
1
1
3
[(9+8j), 'Coding', 'Coding', 'Coding']
['Coding', 'Coding', 'Coding', (9+8j)]


In [41]:
# Sort in asending order: 
b3 = [9,4,13,1,48,3,7,5]
b3.sort()
print(b3)
# sort in descending order: it is use to return list in descending order. 
b3.sort(reverse=True)
print(b3)
# clear: return empty list. delete all element in the list.
# b3.clear()
# print(b3)
# copy: copy's list in another destination.
b4 = b3.copy()
print(b4)

[1, 3, 4, 5, 7, 9, 13, 48]
[48, 13, 9, 7, 5, 4, 3, 1]
[48, 13, 9, 7, 5, 4, 3, 1]


In [2]:
# append
a = [1,1.2,2+3j,"Python","Coding","Coding","Coding"] 
a.append("Pratik")
print(a)

# extend
a.extend("Bye")
print(a)

# insert
a.insert(3,5)
print(a)

# remove
a.remove("Python")
print(a)

# pop
a.pop(3)
print(a)

# index
print(a.index(1))

# count
print(a.count("Coding"))

# reverse
a.reverse()
print(a)

# sort -> only Numbers in ascending order
b =[1,2,3,4,5,6,7,8,9,10]
b.sort() 
print(b)

# sort in reverse -> reverse=True
b.sort(reverse=True)
print(b)

# copy
c = b.copy()
print(c)

[1, 1.2, (2+3j), 'Python', 'Coding', 'Coding', 'Coding', 'Pratik']
[1, 1.2, (2+3j), 'Python', 'Coding', 'Coding', 'Coding', 'Pratik', 'B', 'y', 'e']
[1, 1.2, (2+3j), 5, 'Python', 'Coding', 'Coding', 'Coding', 'Pratik', 'B', 'y', 'e']
[1, 1.2, (2+3j), 5, 'Coding', 'Coding', 'Coding', 'Pratik', 'B', 'y', 'e']
[1, 1.2, (2+3j), 'Coding', 'Coding', 'Coding', 'Pratik', 'B', 'y', 'e']
0
3
['e', 'y', 'B', 'Pratik', 'Coding', 'Coding', 'Coding', (2+3j), 1.2, 1]
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
[10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
[10, 9, 8, 7, 6, 5, 4, 3, 2, 1]


- `Dictionary`

In [4]:
data1 = {1:"Python", 2:"Class",3:4+9j,4:9.3,5:"Code"}
print(data1)

# get
print(data1.get(1))
print(data1.get(4))

# items -> Tuple Format
print(data1.items())

# keys -> Tuple Format
print(data1.keys())

# values -> Tuple Format
print(data1.values())

# a = (1,2,3)
# b = "India"
# data2.fromkeys(a,b)

# copy
data2 = data1.copy()
print(data1)
print(data2)

# update
data1.update({6:"Object"})
print(data1)
data1.update({1:"Guido"})
print(data1)

# pop -> by default remove the last key:value pair.
data1.pop(2)
print(data1)

# by default not remove without key
# data1.pop()
# print(data1)

# popitem -> not popitems
data1.popitem()
print(data1)

# clear
data1.clear()
print(data1)

{1: 'Python', 2: 'Class', 3: (4+9j), 4: 9.3, 5: 'Code'}
Python
9.3
dict_items([(1, 'Python'), (2, 'Class'), (3, (4+9j)), (4, 9.3), (5, 'Code')])
dict_keys([1, 2, 3, 4, 5])
dict_values(['Python', 'Class', (4+9j), 9.3, 'Code'])
{1: 'Python', 2: 'Class', 3: (4+9j), 4: 9.3, 5: 'Code'}
{1: 'Python', 2: 'Class', 3: (4+9j), 4: 9.3, 5: 'Code'}
{1: 'Python', 2: 'Class', 3: (4+9j), 4: 9.3, 5: 'Code', 6: 'Object'}
{1: 'Guido', 2: 'Class', 3: (4+9j), 4: 9.3, 5: 'Code', 6: 'Object'}
{1: 'Guido', 3: (4+9j), 4: 9.3, 5: 'Code', 6: 'Object'}
{1: 'Guido', 3: (4+9j), 4: 9.3, 5: 'Code'}
{}


### `Inheritance: When The Object of child class aquires properties of object of parent class is called Inheritance. When One Child class aquires the properties of parent class is called Inheritance. V25.01.27`
- `It is Object Oriented Concept.`
- `Parent Class: Parent calss|Child Class|Super Class.`
- `Child Class: Child Class|Derived Class|Sub Class.`
- `It is Not mendatory to always say object word in the types of Inheritance defination.`
### `Types of Inheritance:`
- `Single Level Inheritance: When Object of One Child class aquries the properties of object of One Parent class.`
- `Multilevel Inheritance: When One child class is the parent of another child class such inheritance is called as multilevel inheritance`
- `Multiple Inheritance: When One child class inherited the properties of multiple parent class such inheritance is called as Multiple Inheritance.`
- `Hybrid Inheritance: `

In [4]:
# Single Level Inheritance.
class Parent:
    def parent_method(self):
        print("Parent Method.")
class Child(Parent):
    def child_method(self):
        print("Child Method.")
obj = Child()
obj.child_method()
obj.parent_method()
print(Child.mro())
# By default Constructor call in python when object is created.
# Two Types Calling:
# 1. Child() -> by Default -> Implicit 
# 2. obj.Child() -> By Own Hand -> Explicit
# Constructor creates Object(Memory) by default.
# By default Object created Constructor Implictly.

Child Method.
Parent Method.
[<class '__main__.Child'>, <class '__main__.Parent'>, <class 'object'>]


In [3]:
# Multilevel Inheritance.
class Pandu:
    def read(self):
        print("Reading..")
class Pandi(Pandu):
    def code(self):
        print("Coading..")
class Bablu(Pandi):
    def drive(self):
        print("Driving..")
class Babli(Bablu):
    def painting(self):
        print("Painting..")
obj = Babli()
obj.painting()
obj.drive()
obj.code()
obj.read()
print(Babli.mro())
# mro() -> Method Resolution Order. It is nothing but scequence of properties in Inheritance. -> Proritywise.
# Babli, Bunty, Pandi, Pandu, Object.

Painting..
Driving..
Coading..
Reading..
[<class '__main__.Babli'>, <class '__main__.Bablu'>, <class '__main__.Pandi'>, <class '__main__.Pandu'>, <class 'object'>]


In [1]:
# Another Example of Multilevel Inheritance.
class Add:
    def addition(self, num1, num2):
        print("Addition of two Numbers is",num1+num2)
class Sub(Add):
    def subtraction(self, num1, num2):
        print("Subtraction of two Numbers is",num1-num2)
class Mul(Sub):
    def mutiplication(self, num1, num2):
        print("Product of two Numbers is",num1*num2)
class Div(Mul):
    def division(self, num1, num2):
        print("Questiont of two Numbers is",num1/num2)
class Mod(Div):
    def modulos(self, num1, num2):
        print("Remainder of two Numbers is",num1%num2)
num1 = int(input("Enter First Number: "))
num2 = int(input("Enter Second Number: "))
obj = Mod()
obj.modulos(num1,num2)
obj.division(num1,num2)
obj.mutiplication(num1,num2)
obj.subtraction(num1,num2)
obj.addition(num1,num2)
print(Mod.mro())
# Mod, Div, Mul, Sub, Add, Object.

Enter First Number:  5
Enter Second Number:  5


Remainder of two Numbers is 0
Questiont of two Numbers is 1.0
Product of two Numbers is 25
Subtraction of two Numbers is 0
Addition of two Numbers is 10
[<class '__main__.Mod'>, <class '__main__.Div'>, <class '__main__.Mul'>, <class '__main__.Sub'>, <class '__main__.Add'>, <class 'object'>]


In [5]:
# Multiple Inheritance.
class A:
    def m1(self):
        print("A")
class B:
    def m2(self):
        print("B")
class C:
    def m3(self):
        print("C")
class D(A,B,C):
    def m4(self):
        print("D")
obj = D()
obj.m4()
obj.m3()
obj.m2()
obj.m1()
print(D.mro())
# D, A, B, C, Object. 

D
C
B
A
[<class '__main__.D'>, <class '__main__.A'>, <class '__main__.B'>, <class '__main__.C'>, <class 'object'>]


- Diffreance Between Multilevel and Multiple Inheritance.

- `Super Keyword:`
1. It is keyword.
2. Super Keyword Access the Imediate parent class properties.
3. It is Time Efficient.
4. Act's as an object of parent class.

In [8]:
# Accessing the Element of Class B element from Class C and class A element from class B.
class A:
    def m1(self):
        print("A")
class B(A):
    def m2(self):
        print("B")
        super().m1()
        # Call Imediate Parent Class.
class C(B):
    def m3(self):
        print("C")
        super().m2()
        # Call Imediate Parent Class.
obj = C()
obj.m3()
print(D.mro())

C
B
A
[<class '__main__.D'>, <class '__main__.A'>, <class '__main__.B'>, <class '__main__.C'>, <class 'object'>]


In [9]:
# Accessing the Class A Elements from the Class C.
class A:
    def m1(self):
        print("A")
class B(A):
    def m2(self):
        print("B")
class C(B):
    def m3(self):
        print("C")
        super().m1()
        # C->A
        # Call Imediate Parent Class. Still Implimented this Concept Beacouse, 
        # It Accessing the elements of Parent calss B which is inherited from the Parent calss A. 
        # Beacouse of Inheritance Still it can access the m1 method into the B class. 
obj = C()
obj.m3()
print(D.mro())
# C, A, Object.

C
A
[<class '__main__.D'>, <class '__main__.A'>, <class '__main__.B'>, <class '__main__.C'>, <class 'object'>]


- `Call Imediate Parent Class. Still Implimented this Concept Beacouse, It Accessing the elements of Parent calss B which is inherited from the Parent calss A. Beacouse of Inheritance Still it can access the m1 method into the B class.`